iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -328,6 +345,7 @@ static void overlayTableProperties(Configuration configuration, TableDesc tableD
     map.put(InputFormatConfig.TABLE_IDENTIFIER, props.getProperty(Catalogs.NAME));
     map.put(InputFormatConfig.TABLE_LOCATION, table.location());
     map.put(InputFormatConfig.TABLE_SCHEMA, schemaJson);
+    props.put(InputFormatConfig.PARTITION_SPEC, PartitionSpecParser.toJson(table.spec()));
```

Comment:
This is again something which is not by the spec. What happens if this will not propagated?

Reply:
You are right. However, all this code will soon be removed anyway after the table cache PR has been merged, so I would leave it as it is for now if that's OK with you.